src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
I see. Guess that's fine. Support for the kMessageTypeReplace would be a great addition, to allow changes to the track list / queue. This message type seems to be used for any kind of queue manipulation, e.g. changing order of songs in the queue or tapping "add next" in the Android app (they recently added support for this).